azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource_test.go

```diff
@@ -0,0 +1,248 @@
+			return err
+		}
+
+		result, err := client.Get(ctx, id.ResourceGroup, id.Name)
+
+		if err == nil {
+			return nil
+		}
+
+		if result.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Virtual Desktop Host Pool %q (Resource Group: %q) does not exist", id.Name, id.ResourceGroup)
+		}
+
+		return fmt.Errorf("Bad: Get virtualDesktopHostPoolClient: %+v", err)
+	}
+}
+
+func testCheckAzureRMDesktopVirtualizationHostPoolDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).DesktopVirtualization.HostPoolsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_virtual_desktop_host_pool" {
+			continue
+		}
+
+		log.Printf("[WARN] azurerm_virtual_desktop_host_pool still exists in state file.")
+
+		id, err := parse.VirtualDesktopHostPoolID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		result, err := client.Get(ctx, id.ResourceGroup, id.Name)
+
+		if err == nil {
+			return fmt.Errorf("Virtual Desktop Host Pool still exists:\n%#v", result)
+		}
+
+		if result.StatusCode != http.StatusNotFound {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMVirtualDesktopHostPool_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-vdesktop-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_desktop_host_pool" "test" {
+  name                 = "acctestHP%s"
+  location             = azurerm_resource_group.test.location
+  resource_group_name  = azurerm_resource_group.test.name
+  type                 = "Pooled"
+  validate_environment = true
+  load_balancer_type   = "BreadthFirst"
+}
+
+`, data.RandomInteger, data.Locations.Secondary, data.RandomString)
+}
+
+func testAccAzureRMVirtualDesktopHostPool_complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-vdesktop-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_desktop_host_pool" "test" {
+  name                     = "acctestHP%s"
+  location                 = azurerm_resource_group.test.location
+  resource_group_name      = azurerm_resource_group.test.name
+  type                     = "Pooled"
+  friendly_name            = "A Friendly Name!"
+  description              = "A Description!"
+  validate_environment     = true
+  load_balancer_type       = "BreadthFirst"
+  maximum_sessions_allowed = 100
+  preferred_app_group_type = "Desktop"
+
+  # Do not use timestamp() outside of testing due to https://github.com/hashicorp/terraform/issues/22461
+  registration_info {
+    expiration_date = timeadd(timestamp(), "48h")
+  }
+  lifecycle {
+    ignore_changes = [
+      registration_info[0].expiration_date,
+    ]
+  }
+
+  tags = {
+    purpose = "acceptance-testing"
```

Comment:
Could we ensure capitalized tags work?
```suggestion
    Purpose = "Acceptance-Testing"
```